go-controller/pkg/ovn/egressgw.go

```diff
@@ -386,7 +363,7 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod, namespace string) {
 
 	for _, gwIP := range foundGws.gws {
 		// check for previously configured pod routes
-		routeInfos := oc.getRouteInfosForGateway(gwIP.String(), namespace)
+		routeInfos := oc.getRouteInfosForNamespace(namespace)
```

Comment:
the downside to this is you are going to lock more routeInfos unnecessarily. I don't see much value in this 2nd commit.

Reply:
It fixes a real concurrent data access in the original code. There's no way to safely filter the routeInfos how `getRouteInfosForGateway()` was doing without locking each and every routeInfo in that namespace, right? You have to lock to check whether podExternalRoutes[] has the gatewayIP you care about.

Reply:
ah now it makes sense. Thanks.